Sources/Lifecycle/Lifecycle.swift

```diff
@@ -473,9 +473,12 @@ public class ComponentLifecycle: LifecycleTask {
             guard case .idle = self.state else {
                 preconditionFailure("invalid state, \(self.state)")
             }
-            self.log("starting")
             self.state = .starting(queue)
         }
+
+        self.log("starting")
+        Counter(label: "\(self.label).lifecycle.start").increment()
```

Comment:
is label risky i.e. containing spaces etc?

Reply:
hmm good question, do you think we should normalize it?

Reply:
Was just thinking about it but I also just remembered that last time we said that the libs should sanitize, so we do e.g. in the prometheus one https://github.com/MrLotU/SwiftPrometheus/blob/f4d4adfeb1178e250a0ea427017fb4d1cc877de3/Sources/Prometheus/PrometheusMetrics.swift#L112 🤔 


I guess it's fine 👍